test/test_dynamic_shapes.py

```diff
@@ -222,6 +222,161 @@ def test_unsqueeze_copy_dynamism(self):
+    self.assertEqual(t8.cpu(), t8_aten.cpu())
+
+  def test_view_copy_symint_with_dyn_input_static_input_shape(self):
+    # If the input tensor and shape are “statically” incompatible, a compilation error is raised.
+    t1 = torch.tensor([1, 0, 3, 5, 0, 6], device=dev)
+    # t2.shape=torch.Size([<=6, 1]) with real size [4, 1]
+    # t2 = [[0], [2], [3], [5]]
+    t2 = torch.nonzero(t1)
+    # static 6 is incompatible with 7
+    self.assertRaises(RuntimeError, lambda: t2.view(7))
+
+    # If their “dynamic” values are incompatible, a RuntimeError is raised.
+    # statically compatible.
+    # It will fail in pytorch layer.
+    self.assertRaises(RuntimeError, lambda: t2.view(6))
+
+    # verify if dynamism is propagated correctly.
+    t5 = torch.tensor([1, 1, 3, 5, 1, 6], device=dev)
+    t6 = torch.nonzero(t5)
+    t7 = t6.view(6)
+    self.assertIsInstance(t7.shape[0], torch.SymInt)
+    self.assertEqual(str(t7.shape[0]), '<=6')
+    self.assertEqual(t7.shape[0], 6)
+
+    # verify correctness.
+    t5_aten = torch.tensor([1, 1, 3, 5, 1, 6])
+    t6_aten = torch.nonzero(t5_aten)
+    t7_aten = t6_aten.view(6)
+    self.assertEqual(t7.cpu(), t7_aten.cpu())
+
+  def test_view_copy_symint_with_dyn_input_dyn_input_shape(self):
+    # If the input tensor and shape are “statically” incompatible, a compilation error is raised.
+    t1 = torch.tensor([1, 0, 3, 5, 0, 6], device=dev)
+    # t2.shape=torch.Size([<=6, 1]) with real size [4, 1]
+    # t2 = [[0], [2], [3], [5]]
+    t2 = torch.nonzero(t1)
+    t3 = torch.tensor([1, 0, 3, 5, 0, 6, 7], device=dev)
+    # t4.shape=torch.Size([<=7, 1]) with real size [5, 1]
+    t4 = torch.nonzero(t3)
+    self.assertRaises(RuntimeError, lambda: t2.view(t4.shape[0]))
+
+    # If their “dynamic” values are incompatible, a RuntimeError is raised.
+    t5 = torch.tensor([1, 2, 3, 4, 5, 6, 0], device=dev)
+    # t6.shape=torch.Size([<=7, 1]) with real size [6, 1]
+    t6 = torch.nonzero(t5)
+    # statically compatible but dynamically incompatible.
+    # It will fail in pytorch layer.
+    self.assertRaises(RuntimeError, lambda: t6.view(t4.shape[0]))
+
+    # verify if dynamism is propagated correctly.
+    t7 = torch.tensor([1, 0, 3, 5, 0, 6, 7], device=dev)
+    t8 = torch.nonzero(t7)
+    # t8.shape=torch.Size([<=7, 1]) with real size [5, 1]
+    t9 = t8.view(t4.shape[0])
+    self.assertIsInstance(t9.shape[0], torch.SymInt)
+    self.assertEqual(str(t9.shape[0]), '<=7')
+    self.assertEqual(t9.shape[0], 5)
+
+    # verify correctness.
+    t7_aten = torch.tensor([1, 0, 3, 5, 0, 6, 7])
+    t8_aten = torch.nonzero(t7_aten)
+    # t8_aten.size=[5, 1]
+    t3_aten = torch.tensor([1, 0, 3, 5, 0, 6, 7])
+    t4_aten = torch.nonzero(t3_aten)
+    # t4_aten.size=[5, 1]
+    t9_aten = t8_aten.view(t4_aten.shape[0])
+    self.assertEqual(t9.cpu(), t9_aten.cpu())
+
+  def test_view_copy_symint_negative_input_shape(self):
+    t5 = torch.tensor([1, 1, 3, 5, 1, 6], device=dev)
```

Comment:
it is a bit weird that tensor started with t5 lol, can we update that to t1?

Reply:
lol, done